code/modules/mob/living/simple_animal/hostile/megafauna/_megafauna.dm

```diff
@@ -118,24 +118,46 @@
 			if(!client && ranged && ranged_cooldown <= world.time)
 				OpenFire()
 
-			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still gibs yall
+			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH)) //Nope, it still kills yall
 				devour(L)
 		else
 			devour(L)
 
 /// Devours a target and restores health to the megafauna
 /mob/living/simple_animal/hostile/megafauna/proc/devour(mob/living/L)
-	if(!L)
+	if(!L || L.has_status_effect(/datum/status_effect/gutted))
 		return FALSE
-	visible_message(
-		span_danger("[src] devours [L]!"),
-		span_userdanger("You feast on [L], restoring your health!"))
+	celebrate_kill()
 	if(!is_station_level(z) || client) //NPC monsters won't heal while on station
 		adjustBruteLoss(-L.maxHealth/2)
 	L.investigate_log("has been devoured by [src].", INVESTIGATE_DEATHS)
-	L.gib()
+	var/mob/living/carbon/carbonTarget = L
+	if(istype(carbonTarget))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LUNGS))
+		qdel(L.get_organ_slot(ORGAN_SLOT_HEART))
+		qdel(L.get_organ_slot(ORGAN_SLOT_LIVER))
+	L.adjustBruteLoss(500)
+	L.death() //make sure they die
+	L.apply_status_effect(/datum/status_effect/gutted)
+	LoseTarget()
 	return TRUE
 
+/mob/living/simple_animal/hostile/megafauna/proc/celebrate_kill(mob/living/L)
+	visible_message(
+		span_danger("[src] devours [L]'s organs!"),
+		span_userdanger("You feast on [L]'s organs, restoring your health!"))
```

Comment:
```suggestion
		span_danger("[src] disembowels [L]!"),
		span_userdanger("You feast on [L]'s organs, restoring your health!"))
```
I'll be honest I think the specificity of "destroyed organs" in these messages is both unnecessary and also reads a little goofy, but you can leave this one in if you think it is important.